Only check sequence cache for txn writes

tbg opened the pull request:
Fixes #4781. We erroneously checked the sequence cache for all `Write`
commands, but we only need to do it for commands which leave intents.
TestTxnCoordSenderHeartbeat failed spuriously because the transaction
it uses to heartbeat does not update its sequence number (which would
also not happen in practice, at least not instantly).

<!-- Reviewable:start -->

[<img src="https://reviewable.io/review_button.svg" height="40" alt="Review on Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/4795)

<!-- Reviewable:end -->

tamird commented:
LGTM

---

Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[storage/replica_test.go, line 1621 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4795#-KBtNbRb7xhBy7LP6vpW:-KBtNbRcBpJwcXJcI8M3:-1634998954) ([raw file](https://github.com/cockroachdb/cockroach/blob/e13c4ec8074e1859b8e3f5df96b2d85a09916e64/storage/replica_test.go#L1621)):*
do we need a test for the transaction write case? if it already exists, perhaps the two tests should be combined?

---

*[storage/replica_test.go, line 1636 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4795#-KBtNEgpxCzSPEVDbPR0:-KBtNEgpxCzSPEVDbPR1:-1868168467) ([raw file](https://github.com/cockroachdb/cockroach/blob/e13c4ec8074e1859b8e3f5df96b2d85a09916e64/storage/replica_test.go#L1636)):*
what was the error before this change? transaction retry? add a comment.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4795#-:-KBtOFq7f7PGRQGWBpMx:676834233)*
<!-- Sent from Reviewable.io -->

tbg committed:
Only check sequence cache for txn writes

Fixes #4781. We erroneously checked the sequence cache for all `Write`
commands, but we only need to do it for commands which leave intents.
TestTxnCoordSenderHeartbeat failed spuriously because the transaction
it uses to heartbeat does not update its sequence number (which would
also not happen in practice, at least not instantly).

tschottdorf commented:
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[storage/replica_test.go, line 1621 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/4795#-KBtNbRb7xhBy7LP6vpW:-KBtWJL9ij6UpRw9YJms:2082192519) ([raw file](https://github.com/cockroachdb/cockroach/blob/e13c4ec8074e1859b8e3f5df96b2d85a09916e64/storage/replica_test.go#L1621)):*
I tried that initially but decided it was more transparent to not shove more stuff into the existing tests.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/4795)*
<!-- Sent from Reviewable.io -->
